carts/serializers.py

```diff
@@ -0,0 +1,71 @@
+"""
+Serializers for carts models
+"""
+from django.utils.translation import gettext_lazy as _
+from rest_framework import serializers
+
+from carts.models import Cart, CartItem
+
+
+class CartSerializer(serializers.ModelSerializer):
+    """
+    Cart model serializer
+    """
+
+    def create(self, validated_data):
+        validated_data["created_by"] = validated_data["user"]
+        return Cart.objects.create(**validated_data)
+
+    class Meta:
+        """
+        Tells the fields to include in parsed/json object
+        """
+
+        # pylint: disable=protected-access, no-member
+        model = Cart
+        fields = "__all__"
+
+
+class CartItemSerializer(serializers.ModelSerializer):
```

Comment:
What happens if a user deletes an item from the cart of decreases the quantity of an item from the cart?

Reply:
If the user `deletes` an item from the cart this item is `removed` from the `cart`. If quantity is `decreased/updated` there is a check that ensures that the new quantity is at least `1`

Reply:
I meant what will happen to the stock_quantity in the product table? Would that be update accordingly based on items being added and deleted from the cart?

Reply:
No it is not being updated for now. I thought about it and concluded that it would be better to handle that whenever user checkouts a cart or places an order. What do you think about it? Should I change the `stock_quantity` whenever a user adds or removes item from the cart?

Reply:
Yes, our checks on quantity for adding an item should be based on how many items we have in stock. We should maintain the stock quantity.